### PR TITLE
Implement new migration flow for local orders

### PR DESCRIPTION
**Pull request by Naatan**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2506" title="DX-2506" target="_blank">StoryDX-2506</a>  Local order has a backward compatible upgrade mechanic
  </td></table>
  <br />

### CODE REVIEW

File: pkg/projectfile/projectfile.go [hunk 1]
```diff
@@ -564,6 +564,39 @@ func (p *Project) LegacyCommitID() string {
...
+	}
+
+	p.parsedURL.LegacyCommitID = ""
+	p.Project = pf.String()
+	return nil
+}
+
+// SetLegacyCommit sets the commit id within the current project file. This is done
+// in-place so that line order is preserved.
+// If headless is true, the project is defined by a commit-id only
```
Comment (mitchell-as) [inline pkg/projectfile/projectfile.go:584]: Nit: we don't have headless commits anymore, right? So this comment is superfluous.

File: internal/runbits/legacy/projectmigration/projectmigration.go [hunk 2]
```diff
@@ -78,16 +78,110 @@ func PromptAndMigrate(proj projecter) (bool, error) {
...
+	}
+
+	script := projectfile.Script{
+		projectfile.NameVal{
+			Name:  "migrate-to-buildscripts",
+			Value: string(scriptValue),
+		},
+		projectfile.ScriptFields{
+			// No localized since this is a legacy migration mechanic and we'll remove this long before we'll introduce
+			// another language.
```
Comment (mitchell-as) [inline internal/runbits/legacy/projectmigration/projectmigration.go:123]: I'm not following this comment. It looks like there is localization?

**Review by mitchell-as — APPROVED** [review]
Looks good to me. Minor nits to think about and possibly address.

File: internal/runbits/legacy/projectmigration/projectmigration.go (hunk 2 above)
Comment (Naatan) [reply internal/runbits/legacy/projectmigration/projectmigration.go:123]: Forgot to drop it after I moved it to locale cause it was getting noisy.

**Review by mitchell-as — APPROVED** [review]

**Pull request merged**